hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogSyncClient.java

```diff
@@ -138,42 +156,141 @@ public List<Partition> getAllPartitions(String tableName) {
     }
   }
 
+  @Override
+  public List<Partition> getAllPartitions(String tableName) {
+    ExecutorService executorService = Executors.newFixedThreadPool(this.allPartitionsReadParallelism);
```

Comment:
It's quite hard to make unless we want to dig into generics - since the methods I use return void and this is supposed to return a list. So if the code is very simple, I'd prefer it rather than build complex code for 1 method that no one will see